typescript/sdk/src/token/adapters/EvmTokenAdapter.ts

```diff
@@ -251,13 +257,69 @@ export class EvmHypSyntheticAdapter
     return this.getTotalSupply();
   }
 
-  async quoteTransferRemoteGas(
-    destination: Domain,
-  ): Promise<InterchainGasQuote> {
-    const gasPayment = await this.contract.quoteGasPayment(destination);
-    // If EVM hyp contracts eventually support alternative IGP tokens,
-    // this would need to determine the correct token address
-    return { amount: BigInt(gasPayment.toString()) };
+  async quoteTransferRemoteGas({
+    destination,
+    recipient,
+    amount,
+  }: QuoteTransferRemoteParams): Promise<InterchainGasQuote> {
+    const contractVersion = await this.contract.PACKAGE_VERSION();
+
+    const hasQuoteTransferRemote =
+      compareVersions(
+        contractVersion,
+        QUOTE_TRANSFER_REMOTE_CONTRACT_VERSION,
+      ) >= 0;
+
+    // Version does not support quoteTransferRemote defaulting to quoteGasPayment
+    if (!hasQuoteTransferRemote) {
+      const gasPayment = await this.contract.quoteGasPayment(destination);
+      return { igpQuote: { amount: BigInt(gasPayment.toString()) } };
+    }
+
+    assert(
+      !isNullish(amount),
+      'Amount must be defined for quoteTransferRemoteGas',
+    );
+    assert(recipient, 'Recipient must be defined for quoteTransferRemoteGas');
+
+    const recipBytes32 = addressToBytes32(addressToByteHexString(recipient));
+
+    const [igpQuote, ...feeQuotes] = await this.contract.quoteTransferRemote(
+      destination,
+      recipBytes32,
+      amount,
+    );
+
+    const [igpTokenAddressOrDenom, igpAmount] = igpQuote;
+
+    const tokenFeeQuotes: Quote[] = feeQuotes.map((quote) => ({
+      addressOrDenom: quote[0],
+      amount: BigInt(quote[1].toString()),
+    }));
+
+    // Because the amount is added on of the fees we need to subtract it
+    // from the actual IGP/fees
+    let igpBigIntAmount = BigInt(igpAmount.toString());
+    let tokenFeeQuote: Quote | undefined = undefined;
+
+    // For native routes, the fee is returned as a single quote
+    if (!tokenFeeQuotes.length) {
+      igpBigIntAmount = igpBigIntAmount - amount;
```

Comment:
`igpBigIntAmount = igp + native token fee + amount`. so what you end up here is `igp + native token fee`. In other words, you will have to subtract the token fee here to get the igp